includes/forms/class-llms-form-field.php

```diff
@@ -148,6 +148,52 @@ protected function classes_ensure_array( $classes, $defaults = array() ) {
 
 	}
 
+	/**
+	 * Returns an array of form field objects from this checkbox or radio field's options array.
+	 *
+	 * @since [version] Moved from `LLMS_Form_Field::get_field_html()` and added the hidden logic.
+	 *
+	 * @param string $is_hidden If true, returns only the checked fields and sets their type to 'hidden',
+	 *                          else returns all options as `$this->settings['type']` form fields.
+	 * @return LLMS_Form_Field[]
+	 */
+	public function explode_options_to_fields( $is_hidden = false ) {
+
+		$fields = array();
+		$value  = ! empty( $this->settings['value'] ) || is_array( $this->settings['value'] )
+			? $this->settings['value']
+			: $this->settings['default'];
+
+		foreach ( $this->settings['options'] as $key => $val ) {
+
+			$name    = $this->settings['name'];
+			$checked = $value === $key;
+
+			if ( 'checkbox' === $this->settings['type'] ) {
+				$name    .= '[]';
+				$value   = is_array( $value ) ? $value : array( $value );
+				$checked = in_array( $key, $value, true );
+			}
+
+			if ( $is_hidden && false === $checked ) {
+				continue;
+			}
+
+			$fields[] = new LLMS_Form_Field(
```

Comment:
this was
```suggestion
			$fields[] = new self(
```
is there a reason why we prefer this other way? Just to align myself with cs.

Reply:
At one point I moved this code to another class, which required the specific class name. When I moved it back, I neglected to change the class name back to `self`, which I prefer.